File: Behavioral/Lab22/Typescript/src/program.ts

import { AbstractTemplate } from './TemplateMethods/abstract_template';
import { AddApartmentTemplate } from './TemplateMethods/add_apartment_template';
import { AddHouseStrategy } from './TemplateMethods/add_house_template';
import { AddRoomStrategy } from './TemplateMethods/add_room_template';
class Program {
    private static _addStrategy : AbstractTemplate
    static main(args: string[]): void {
        console.log("Start testing... \n\n");
        [   new AddApartmentTemplate(),
            new AddRoomStrategy("room", "room", "1", "1", "test"),
            new AddHouseStrategy("house", "house", "1", "1", "test"),
            new AddHouseStrategy("house", "house", "1", "1",),
            new AddRoomStrategy("error"),]
        .forEach((strategy: AbstractTemplate) => {
            this._addStrategy = strategy;
            this.executeCommands();
        });
        console.log("End testing... \n\n");
    }

    private static executeCommands(): void {
        console.log(this._addStrategy.execute());
    }
}

Program.main([]);

File: Behavioral/Lab22/C#/Program.cs
using System;
using System.Diagnostics;
using System.Net.WebSockets;
using System.Reflection.Emit;
using System.Runtime.InteropServices;
using C_.Builders;
using C_.TemplateMethods;
using C_.Directors;
using C_.Entities;

namespace C_
{    
    class Program
    {
        private static AbstractTemplate _addTemplate;

        static void Main(string[] args)
        {
            Console.WriteLine("Start testing... \n\n");
            foreach (AbstractTemplate strategy in new List<AbstractTemplate>{
                new AddApartmentTemplate(),
                new AddRoomTemplate("test", "test", "1", "1", "test"),
                new AddHouseTemplate("house", "house", "1", "1", "test"),
                new AddHouseTemplate("house", "house", "1", "1"),
                new AddRoomTemplate("error"),})
            {
                _addTemplate = strategy;
                AddEstate();
            };
            Console.WriteLine("End testing... \n\n");
        }

        private static void AddEstate(){
            Console.WriteLine(_addTemplate.Execute());
        }

    }
}
